src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -379,6 +380,18 @@ function VisualizeAppController(
       },
     };
 
+    const handleLinkedSearch = linked => {
+      if (linked && !savedVis.savedSearchId) {
```

Comment:
this confuses me ... you desctructure `savedSearchId`  from `savedVis` ... so this logic is actually:
```
if (linked && !savedVis.savdSearchId) {
  savedVis.savedSearchId = savedVis.savedSearchId;
  vis.savedSearchId = savedVis.savedSearchId;
  ...
}
```

also i am not sure why we need to duplicate information on `vis.savedSearchId`, which is also violating the types (vis has no savedSearchId property)

Reply:
i think once you unlink, all we need is to delete savedVis.savedSearchId 
and when linking back, you will need to select the saved search, so you get the new savedSearchId and you can set it on savedVis

Reply:
ps: i am working on a small refactor to vis saved object and vis state, which will allow to further clean this up: you will no longer need to care about savedVis here, all you will need to do is remove parent search source and we will provide utilities to serialize searchSource + vis into the saved vis object.

Reply:
Hey @ppisljar , thanks for taking a look and reasonable comments here!
PR updated!
About storing the `vis.savedSearchId` , I'm not sure why, but it's presented in the `vis` object on creation of a visualization based on saved search:

![image](https://user-images.githubusercontent.com/31325372/76413827-34845f00-63a7-11ea-8a74-87b5e4a0d197.png)

so would be great if you could get rid of it in your refactoring pr.